clients/unshaded/src/main/java/tachyon/client/TachyonFS.java

```diff
@@ -97,17 +97,17 @@ public static synchronized TachyonFS get(final TachyonURI tachyonURI) throws IOE
    */
   public static synchronized TachyonFS get(final TachyonURI tachyonURI, TachyonConf tachyonConf)
       throws IOException {
-    Preconditions.checkNotNull(tachyonConf, "Could not pass null TachyonConf instance.");
-    if (tachyonURI == null) {
-      throw new IOException("Tachyon Uri cannot be null. Use " + Constants.HEADER + "host:port/ ,"
-          + Constants.HEADER_FT + "host:port/");
-    }
+    Preconditions.checkNotNull(tachyonConf, "Could not pass a null TachyonConf instance.");
+    Preconditions.checkNotNull(tachyonURI, "Tachyon URI cannot be null. Use " + Constants.HEADER
+        + "host:port/ ," + Constants.HEADER_FT + "host:port/.");
     String scheme = tachyonURI.getScheme();
-    if (scheme == null || tachyonURI.getHost() == null || tachyonURI.getPort() == -1
-        || (!Constants.SCHEME.equals(scheme) && !Constants.SCHEME_FT.equals(scheme))) {
-      throw new IOException("Invalid Tachyon URI: " + tachyonURI + ". Use " + Constants.HEADER
-          + "host:port/ ," + Constants.HEADER_FT + "host:port/");
-    }
+    Preconditions.checkNotNull(scheme, "Tachyon scheme cannot be null. Use tachyon:// "
+        + "or tachyon-ft://.");
+    Preconditions.checkNotNull(tachyonURI.getHost(), "Tachyon hostname cannot be null.");
+    Preconditions.checkArgument(tachyonURI.getPort() != -1, "Tachyon URI must have a port number.");
```

Comment:
Maybe this should be checkState(), since is not directly checking the method argument.